src/app/hooks/useAccountMetadata.ts

```diff
@@ -24,7 +26,17 @@ export const useAccountMetadata = (scope: SearchScope, address: string): Account
     enabled: !isPontusX && !isLocalnet(scope.network),
     useErrorBoundary: false,
   })
-  return isPontusX ? pontusXData : oasisData
+  const registryData = isPontusX ? pontusXData : oasisData
+
+  // Also look up self-professed metadata (for tokens)
+  const { token, isLoading: isTokenLoading, isError: isTokenError } = useTokenInfo(scope, address)
```

Comment:
This gets loaded often; could cache for longer

https://pr-1773.oasis-explorer.pages.dev/mainnet/sapphire/address/0x5f88516FB09575F961f81f2b8BCE38ff5e790750?page=3
click next page
loads Oasis Punks token again

Reply:
Added rudimentary caching.

Reply:
OK, added react-query caching.